### DISCUSSION TITLE
What exactly is the purpose of document_id, doc_id, and ref_doc_id in nodeToMetadata when they all point to the same value?

### DISCUSSION THREAD

**Discussion by shodevacc** [run-llama/LlamaIndexTS#1101] [Q&A]
When ingesting documents using an Ingestion pipeline is there a reason why there are three references to node.sourceNode?.nodeId being stored as metadata? What purpose could they serve if they reference the same value?

I'm referring to [these lines](https://github.com/run-llama/LlamaIndexTS/blob/ba058dc8d4276912b4f1850c60b3668159ada7d8/packages/llamaindex/src/storage/vectorStore/utils.ts#L34-L36) in the `packages/llamaindex/src/storage/vectorStore/utils.ts` file:

```typescript
  metadata["document_id"] = node.sourceNode?.nodeId || "None";
  metadata["doc_id"] = node.sourceNode?.nodeId || "None";
  metadata["ref_doc_id"] = node.sourceNode?.nodeId || "None";
```

**Comment by marcusschiesser** [comment]
AFAIK they are there for compatibility reasons with older Python versions. @logan-markewich, correct?
